spec/hardware/iommu.fmf

```diff
@@ -0,0 +1,26 @@
+summary: |
+    Select or provision a guest with the `Input–output memory management unit`.
+
+description: |
+    .. code-block::
+
+       iommu:
+           # Boolean, whether the guest supports IOMMU
+           is-supported: "x"
+           # String, the model name for virtualizated IOMMU
+           model: "virtio"|"intel"|"smmuv3"
```

Comment:
First, I wouldn't say "virtualized". The model is simply the one requested - it may be satisfied by an actual bare-metal machine, correct? `virtual` plugin may spawn a VM with the requested IOMMU model, but `beaker` may be able to find a machine with the specified model.

Second, I would probably drop the list from specification, or at least make it an open-ended list, see https://github.com/teemtee/tmt/blob/main/spec/hardware/virtualization.fmf#L15. There may be more models or aliases, let's not suggests these are the only ones possible.

Reply:
> First, I wouldn't say "virtualized". The model is simply the one requested - it may be satisfied by an actual bare-metal machine, correct? `virtual` plugin may spawn a VM with the requested IOMMU model, but `beaker` may be able to find a machine with the specified model.

Here I intended to say: the model is only for the plugins like `virtual` or `artemis.openstack`. It will not work for `beaker`. Is it OK to comment that here?
> 
> Second, I would probably drop the list from specification, or at least make it an open-ended list, see https://github.com/teemtee/tmt/blob/main/spec/hardware/virtualization.fmf#L15. There may be more models or aliases, let's not suggests these are the only ones possible.

Good idea~

Reply:
> > First, I wouldn't say "virtualized". The model is simply the one requested - it may be satisfied by an actual bare-metal machine, correct? `virtual` plugin may spawn a VM with the requested IOMMU model, but `beaker` may be able to find a machine with the specified model.
> 
> Here I intended to say: the model is only for the plugins like `virtual` or `artemis.openstack`. It will not work for `beaker`. Is it OK to comment that here?

I don't think so, because AFAICT, it's not limited to `virtual` or `openstack`, AWS and Azure can also deliver machines with IOMMU support, and I bet the list continues. And they may use the same model names, or even add new ones, so the `model` key is not that pinned to these plugins only.

In general, the specification should remain infrastructure-agnostic, especially when it comes to genuine HW requirements. It's slightly different when speaking about not-so-much-HW-but-still-affecting-provisioning requirements, like job delegation in Beaker, as no other infrastructure has this particular feature, but IOMMU is not that specific. It can be delivered by multiple infrastructures, at least in theory, we don't have support for it yet, but it's possible, and therefore it's not limited to just `virtual` and `beaker`.

We use `link` to point to plugins that actually implement and support the requirement, see e.g. https://github.com/teemtee/tmt/blob/main/spec/hardware/hostname.fmf#L23. For now, this list would be empty, the patch adding implementation to `beaker` plugin would also update this list with `implemented-by` link, and then other plugins and so on.